plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,331 @@
+package com.mapbox.maps.plugin.viewport
+
+import android.animation.ValueAnimator
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.plugin.animation.CameraAnimationsLifecycleListener
+import com.mapbox.maps.plugin.animation.CameraAnimationsPlugin
+import com.mapbox.maps.plugin.animation.CameraAnimatorType
+import com.mapbox.maps.plugin.animation.Cancelable
+import com.mapbox.maps.plugin.animation.MapAnimationOwnerRegistry
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.viewport.data.DefaultViewportTransitionOptions
+import com.mapbox.maps.plugin.viewport.data.FollowingViewportStateOptions
+import com.mapbox.maps.plugin.viewport.data.OverviewViewportStateOptions
+import com.mapbox.maps.plugin.viewport.data.ViewportOptions
+import com.mapbox.maps.plugin.viewport.state.FollowingViewportState
+import com.mapbox.maps.plugin.viewport.state.FollowingViewportStateImpl
+import com.mapbox.maps.plugin.viewport.state.OverviewViewportState
+import com.mapbox.maps.plugin.viewport.state.OverviewViewportStateImpl
+import com.mapbox.maps.plugin.viewport.state.ViewportState
+import com.mapbox.maps.plugin.viewport.transition.DefaultViewportTransition
+import com.mapbox.maps.plugin.viewport.transition.DefaultViewportTransitionImpl
+import com.mapbox.maps.plugin.viewport.transition.ImmediateViewportTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportTransition
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * Mapbox default implementation for [ViewportPlugin].
+ *
+ * The Viewport plugin allows to track objects on a map.
+ *
+ * It provides a structured approach to organizing camera management logic into states and transitions between them.
+ *
+ * at any given time, the viewport is either:
+ *  - idle (not updating the camera)
+ *  - in a state (camera is being managed by a ViewportState)
+ *  - transitioning (camera is being managed by a ViewportTransition)
+ */
+@MapboxExperimental
+class ViewportPluginImpl : ViewportPlugin {
```

Comment:
shouldn't our impl be `internal`? I guess we are trying to following such logic for other plugins

Reply:
in that case it's better to mark `ViewportPlugin` interface as experimental and do not explicitly mark it here

Reply:
if you remember, we have a compile dependency to plugins on the sdk module to initialise the plugin with concrete implementation, so we have to keep it public.